doc/source/guidelines/doc_practices.rst

```diff
@@ -0,0 +1,352 @@
+   again to ensure that the final red line indicates that these files have been deleted.
+
+#. If you make any additional file changes, type ``git add .`` again to restage the files to
+   commit.
+
+#. When finished, commit your changes to GitHub history by entering ``git commit -m`` followed by
+   a description in quotation marks.
+   
+   For the given example, you might enter:
+   
+   .. code::
+
+    git commit -m "Edit Contributing topic" 
+   
+   The number of files changed are shown, followed by the number of insertions and deletions.
+   A create mode or rename doc action is then shown for each file.
+
+#. Enter ``git push``, which shows a message like this to indicate that changed files have not
+   been uploaded yet:
+   
+   .. code::
+   
+    fatal: The current branch doc/edit_contributing has no upstream branch.
+
+#. To push the  current branch and set the remote as upstream, enter the following,
+   replacing the branch name shown with your branch name:
+   
+   .. code::
+   
+    git push --set-upstream origin doc/edit_contributing
+
+#. Copy the text shown and paste it into the command line and run it.
+
+#. Complete the PR as described in :ref:`complete_pr`.
+
+Using Git Extensions to Push Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This procedure assumes that you are using Git Extensions to push your changes to the
+GitHub repository.
+
+#. In Git Extensions, select ``Commands > Create branch`` and then provide a name for
+   the branch that you want to create, using the prefix ``doc`` followed by a short
+   description of what you intend to change.
+   
+#. Make your documentation changes in the directory where you cloned the library's
+   GitHub reposity.   
+
+#. Test your changes in local documentation builds. For more information, see
+   :ref:`doc_building`.
+
+#. In Git Extensions, click ``Commit`` to see the files that have been changed in your
+   local directory.
+   
+#. In the window that opens, stage the files to commit, add a commit message, and then
+   click ``Commit``.
+
+#. Do a pull from the GitHub repository to your local directory to ensure that there
+   are no conflicts with what is in your branch. If there are conflicts, resolve
+   them.
+
+#. Do a push of your branch to the GitHub repsitory.
+
+#. Complete the PR as described in :ref:`complete_pr`.
+
+.. _complete_pr:
+
+Completing the GetHub PR
+-------------------------
+#. Go to the GitHub reposity, where you will see an entry for your pushed branch.
+
+#. If the pushed branch resolves an issue, in its description, type ``Resolves #``
+   and select from the list of issues that appears.
+
+#. Create the PR, making title a simple statement because this title is rendered
+   in our `automatic release generator <https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes>`_.
+
+#. If you are not ready to have someone review the PR, change its title to include
```

Comment:
I think we should promote the "draft PR" feature for this:

![image](https://user-images.githubusercontent.com/5692808/150490932-3d42c211-bc96-46a4-bcca-24caeada1952.png)

Reply:
@greschd I believe I incorporated all of your review comments but this one, primarily because I don't understand the advantage since this PR is only for doc and can't be merged until it is approved anyhow.